### DISCUSSION TITLE
Address already in use port 53

### DISCUSSION THREAD

**Discussion by Ne-oL** [AdguardTeam/AdGuardHome#5937] [Q&A]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux/ARMv7

### Installation

GitHub releases or script from README

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

0.107.21-1

### Action

i want to mention that the issue was raised by another user, however, it was closed with no satisfactory answer. 
I installed Adguard from the LuCi packages repository. its over a fresh install of OpenWRT. however, once in the initial setup, you get the error of Address already in use. it goes to both the webUI at 80 and DNS at 53. the wizard suggests using 8080 for the UI, however from the DNS server, it tells you to follow the instructions on another page. which discusses the dnsmasq and many other thing that i don't know. i barely have enough knowledge to set openwrt up let alone everything else being discussed in the guide, and I know I'm not alone in this struggle. would it be possible to change the port from 53 to any other port that is not being used? And if it can be changed, wouldn't it be better to change the default from 53? if not, i have seen a Chinese fork that automates everything, would it be possible to merge their automation into the main repository to make it easier for the users?

### Expected result

no "Address already in use" error for setting the initial DNS at port 53

### Actual result

"Address already in use" for DNS at port 53

**Comment by junland** [comment]
This is on the Docker / Podman network side, so basically what you need todo is just set the DNS port to something other than 53. I don't know how to do it for Docker but for Podman just edit `/etc/containers/containers.conf` in the `network` section.

```
[network]
...
# The port used for dns forwarding daemon.
dns_bind_port = 54
...
```

**Comment by ainar-g** [comment]
Something else is running on port 53.  As the [FAQ](https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#bindinuse) says, you can generally use:

```sh
sudo lsof -i :53
```

(or a similar `netstat` command) to see which process occupies the port and handle it.

**Reply by Ne-oL** [reply]
The goal that i wanted to highlight with this issue isn't how to solve it, rather why it exists in the first place? If I'm using a fresh install of openwrt and installing Adguard home, why do i have to jump through all those hoops to get it working? Btw, in the FAQs it mentions that it might show the automatic "fix it" along the error. For me it didn't show it. So as a novice user who managed to install openwrt and wanted to use Adguard, this proved to me a very hard task. Can Adguard home use any port by default, so the user would have a smooth installation process? Lastly, i have no idea where to even run the linux command for netstat, there is no terminal in my router, just LuCi interface.

**Reply by ppfeufer** [reply]
`openwrt` is a router OS which comes with its own built-in DNS resolver. These resolvers listen per default on port 53, as this port is the "industry standard" for DNS resolvers to listen to and DNS clients expect a DNS resolver to be listening on this port.

By installing a DNS proxy (AGH for example), you have to ensure your system's DNS resolver is not running, because it's replaced by the DNS proxy, otherwise, they will fight over this port. So changing which service listens to port 53 is always manual labour.

And I'm pretty sure `openwrt` also has an SSH service running, so you should be able to connect to your router via SSH.

How did you install AGH using the install script if you don't have a console to your router in the first place?

**Reply by Ne-oL** [reply]
I see. So basically our usage is the abnormal one. I was thinking that AGH was exclusive to the openwrt, this assumption is based on my answer to your second question. I downloaded AGH from the package repository in openwrt, a simple click to install process tbh. so i was thinking that since it was forked to this platform, why does it need modifications to work. But your answer clarifies things to me. Regarding the ssh, i found it after your comment and successfully connected with the router but most sudo commands didn't work, I'm still looking for the reasons. One last thing, assuming i was successful in stopping the DNS resolver of my router and installed AGH. then in the future, if i were to remove AGH, would my router dns kick it automatically or will it need restarting manually in a long process?

**Reply by ppfeufer** [reply]
No, AGH is not exclusive to `openwrt`. I for example am running it on a Raspi with Debian 11.

> then in the future, if i were to remove AGH, would my router dns kick it automatically or will it need restarting manually in a long process?

Changing services will be a manual process.
